src/Tizen.MachineLearning.Train/Tizen.MachineLearning.Train/Layer.cs

```diff
@@ -0,0 +1,134 @@
+/*
+* Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+using static Interop;
+using System;
+using System.IO;
+
+namespace Tizen.MachineLearning.Train
+{
+    /// <summary>
+    /// Creates a neural network layer.
+    /// </summary>
+    /// <remarks>
+    /// Use this function to create neural network layer.
+    /// If the function succeeds, layer must be released using Destroy(), if not added to a model.
+    /// If added to a model, layer is available until the model is released.
+    /// </remarks>
+    /// <since_tizen> 10 </since_tizen>
+    public class Layer: IDisposable
+    {
+        private IntPtr handle = IntPtr.Zero;
+        private bool disposed = false;
+
+        /// if true, model will be destroy layer handle
+        private bool notDestroy = false;
+
+        /// <summary>
+        /// Creates a neural network layer.
+        /// </summary>
+        /// <param name="type">The nntrainer layer type.</param>
+        /// <since_tizen> 10 </since_tizen>
+        public Layer(NNTrainerLayerType type)
+        {
+            NNTrainerError ret = Interop.Layer.Create(out handle, type);
+            NNTrainer.CheckException(ret, "Failed to create model instance");
+            Log.Info(NNTrainer.Tag, $"Create layer with type:{type}");
+        }
+
+        internal Layer(bool createdByModel)
+        {
+            notDestroy = createdByModel;
+        }
```

Comment:
`createdByModel` 이름이 적절하지 않는것 같습니다. 

다른 API를 참조하면 
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle.-ctor?view=net-6.0#system-runtime-interopservices-safehandle-ctor(system-intptr-system-boolean)
`ownsHandle`또는 `hasOwnership` 정도가 적당할것 같습니다.

그리고 Layer객체가 사용하는 handle을 생성자를 통해서 전달하는것이 객체의 consistency를 위해서라도 더 좋은것 같습니다.
```suggestion
        internal Layer(IntPtr handle, bool hasOwnership)
        {
            this.hasOwnership = hasOwnership;
            this.handle = handle;
        }
```

사용되는 쪽에서도 아래와 같이 사용되는것이 더 간결하고 가독성이 높을것 같습니다.
``` c#
            Layer layer = new Layer(layerHandle, false);
```

Reply:
GetLayer() 에서 Layer를 생성을 할 때, 제안 하신것 처럼 handle을 받는 것으로 수정하겠습니다.
hasOwnership 라는 용어로 바꾸겠습니다.

Reply:
수정하였습니다. (https://github.com/Samsung/TizenFX/pull/4156/commits/11d39ece5c285e3facee7672c9590724ea941996)